zend_abstract_interface/sandbox/sandbox.h

```diff
@@ -218,8 +220,25 @@ inline bool zai_sandbox_timed_out(void) {
     return false;
 }
 
+inline bool zai_is_request_blocked(void)
+{
+    if (!PG(last_error_message)) {
+        return false;
+    }
+
+    if (strcmp("Datadog blocked the request and presented a static error page", ZSTR_VAL(PG(last_error_message))) == 0) {
```

Comment:
My main concern with this approach is that if someone changes the error message in appsec, it would stop working. Can you make changes on the appsec side to:
- Make sure the error messages are validated at compile time?
- Add comments specifying that the error message must not be changed.